### ISSUE TITLE
getChromInfoFromUCSC: anyNA(m32) is not TRUE error for hg38

### ISSUE THREAD

**Issue by chrarnold** [Bioconductor/GenomeInfoDb#82]
Today, I receive the following error with hg38 specifcally, hg19 and mm10 work as before. Any idea what is causing this? Must be a change in the last few days, it worked always before up until Friday when I last tested.

```
> GenomeInfoDb::getChromInfoFromUCSC("hg38")
Error in .order_seqlevels(chrom_sizes[, "chrom"]) :
!anyNA(m32) is not TRUE
```

GenomeInfoDb version: '1.34.3'

**Comment by mxw010** [comment]
based on the last fix, it's something on ucsc's website. I ran into the same error with hg38 *today* with another package (chipPeakanno), so it's mostly likely some outside changes that caused the error.

**Comment by chrarnold** [comment]
I see ok! Is there a place where this can be reported? Doesnt seem to be associated with any single package. I currently cant build my package because of that, will think about a backup solution until this is fixed.

**Comment by mxw010** [comment]
I googled a bit more, and the last time the bug appeared, chipPeakAnno author suggested updating GenomeInfoDb. I think they used this package as well. I think attaching an output from SessionInfo() would be helpful to the authors of this package to debug.

I think this is likely a bug from outside because I did not have any problem running it last week. This bug re-appeared today.

**Comment by chrarnold** [comment]
Same here, worked Friday and no changes, worked all the time before... hopefully not difficult to fix

**Comment by Shts2123** [comment]
same issue here. It was working last week but now I'm getting the same error.

**Comment by hpages** [comment]
So it looks like UCSC has just sneakily changed the assembly that hg38 is based on, _again_! Used to be GRCh38.p13, now it's GRCh38.p14. I've not seen any announcement. Unfortunately this breaks `GenomeInfoDb::getChromInfoFromUCSC("hg38")`.

Not the first time they do this: see issue #30. They've even done a change to hg19 about 3 years ago: see issue #9 :cry:

A fix is on the way. Will only be available in current BioC release (3.16) and devel (3.17) because past releases are frozen and no longer maintained. So, if you are using a past release (@mxw010 you seem to be in that situation), you'll need to update to the current release.

**Comment by priyatamapandey** [comment]
I ran the annotatr package for hg38 and it returned same error. I don't understand, how can I fixed it.

Thank you for any suggestion.
Priya

**Comment by hpages** [comment]
Fixed in **GenomeInfoDb** 1.34.8 (release) and 1.35.14 (devel). See commit 091b5d2e8ef23ae5bbe915a33d62f66f4ad32585

Note that these new version will take a couple of days to propagate to the Bioconductor repositories and become available via `BiocManager::install()`. Again, only users of BioC 3.16 or 3.17 will be able to update. If you cannot wait, you can install directly from GitHub (with `BiocManager::install("Bioconductor/GenomeInfoDb)"`) but please be aware that you should never do this in normal times, so this should be a one-time exception.

Best,
H.

**Comment by chrarnold** [comment]
Thanks Herves! My package requires this to work, which also means that with the (permanent I guess) change at UCSC, all people who do not use the current Bioc version (i.e., those who use 3.15 or below) wont be able to use the package anymore, as the error will appear for all versions below 1.34.8 correct? I may need to include the GenomeInfoDb dependency to 1.34.8 at least explicitly in my DESCRIPTION from now onwards, for example.

**Comment by chrarnold** [comment]
> Thanks Herves! My package requires this to work, which also means that with the (permanent I guess) change at UCSC, all people who do not use the current Bioc version (i.e., those who use 3.15 or below) wont be able to use the package anymore, as the error will appear for all versions below 1.34.8 correct? I may need to include the GenomeInfoDb dependency to 1.34.8 at least explicitly in my DESCRIPTION from now onwards, for example.



> Fixed in **GenomeInfoDb** 1.34.8 (release) and 1.35.14 (devel). See commit [091b5d2](https://github.com/Bioconductor/GenomeInfoDb/commit/091b5d2e8ef23ae5bbe915a33d62f66f4ad32585)
> 
> Note that these new versions will take a couple of days to propagate to the Bioconductor repositories and become available via `BiocManager::install()`. Again, only users of BioC 3.16 or 3.17 will be able to update. If you cannot wait, you can install directly from GitHub (with `BiocManager::install("Bioconductor/GenomeInfoDb)"`) but please be aware that you should never do this in normal times, so this should be a one-time exception.
> 
> Best, H.

Thanks Herves! My package requires this to work, which also means that with the (permanent I guess) change at UCSC, all people who do not use the current Bioc version (i.e., those who use 3.15 or below) wont be able to use the package anymore, as the error will appear for all versions below 1.34.8 correct? I may need to include the GenomeInfoDb dependency to 1.34.8 at least explicitly in my DESCRIPTION from now onwards, for example.

**Comment by hpages** [comment]
@chrarnold 

> all people who do not use the current Bioc version (i.e., those who use 3.15 or below) wont be able to use the package anymore

That's the case, unfortunately.

> I may need to include the GenomeInfoDb dependency to 1.34.8 at least explicitly in my `DESCRIPTION` from now onwards, for example.

Yep, that's a good idea. You can include the `GenomeInfoDb (>= 1.34.8)` dep in the RELEASE_3_16 branch of your package and the `GenomeInfoDb (>= 1.35.14)` in its devel branch.

H.

**Comment by priyatamapandey** [comment]
HI, Thank you for  explaining the version but I don't see 1.34.8 for GenomeInfoDb. 
https://www.bioconductor.org/packages/release/bioc/html/GenomeInfoDb.html. This is 1.34.7. 
Also, with the latest biocManager, I am not able to install the annotatr package.

Thank you for your help
Priya

**Comment by mjsteinbaugh** [comment]
@priyatamapandey See @hpages response above:

```
Note that these new versions will take a couple of days to propagate to the Bioconductor repositories and become available
```

**Comment by hpages** [comment]
Finally announced today (7 min ago): https://groups.google.com/a/soe.ucsc.edu/g/genome-announce/c/SytF4qkgpMw

Not a word about the 2 extra sequences that they kept from GRCh38.p13 and that don't belong
to GRCh38.p14. See my long commit comment for the details: 091b5d2e8ef23ae5bbe915a33d62f66f4ad32585

**Comment by miachom** [comment]
Hi, I installed directly from GitHub (with BiocManager::install("Bioconductor/GenomeInfoDb)"). But now I am getting error as:
Error in stop_if(is.null(NCBI_assembly_info), "\"assembly_accession\" field in 'NCBI_LINKER' must ",  : 
  Error in UCSC genome registration file 'hg38.R':
  "assembly_accession" field in 'NCBI_LINKER' must be associated with a registered NCBI assembly

Could you please help with this? @hpages

**Comment by chrarnold** [comment]
> Hi, I installed directly from GitHub (with BiocManager::install("Bioconductor/GenomeInfoDb)"). But now I am getting error as: Error in stop_if(is.null(NCBI_assembly_info), ""assembly_accession" field in 'NCBI_LINKER' must ", : Error in UCSC genome registration file 'hg38.R': "assembly_accession" field in 'NCBI_LINKER' must be associated with a registered NCBI assembly
> 
> Could you please help with this? @hpages

I remember seeing the same, but the message disappeared for me after restarting the session and re-loading the newest package version.

**Comment by miachom** [comment]
> Hi, thanks a lot! It worked after restarting!

**Comment by zhiliqiao** [comment]
I ran into the same issue, and it turns out I cannot update the GenomeInfoDb package to the required version (1.34.8). I'm using the `BiocManager::install("Bioconductor/GenomeInfoDb")` command as suggested. My Bioconductor is version 3.16, BiocManager 1.30.19. But what I got from this installation was GenomeInfoDb of version 1.34.4. Removing and reinstalling didn't fix this problem. Can anyone tell me where I did wrong?

**Comment by hpages** [comment]
@zhiliqiao The latest **GenomeInfoDb** has finally propagated to the Bioconductor public repositories so there's no need to install it from GitHub. Just install it the normal way i.e. with `BiocManager::install("GenomeInfoDb")`.

**Comment by AntoniaChroni** [comment]
I did what @hpages suggested but I still get this error:

Error in .order_seqlevels(chrom_sizes[, "chrom"]) : 
  !anyNA(m32) is not TRUE
> rm(chrom.assay) # To clean up the generated chrom.assay object within the for loop.
Warning message:
In rm(chrom.assay) : object 'chrom.assay' not found

Any thoughts, or workarounds on this?

**Comment by hpages** [comment]
@AntoniaChroni So you are using the latest **GenomeInfoDb** (1.34.9 in BioC 3.16 and 1.35.15 in BioC 3.17) and you're still getting this error? What's your `sessionInfo()`? What version of Bioconductor (`BiocManager::version()`) do you use?

As explained above, the fix is only in **GenomeInfoDb** >= 1.34.8 (in BioC 3.16) and **GenomeInfoDb** >= 1.35.14 (in BioC 3.17).

**Comment by AntoniaChroni** [comment]
So that now worked but when I tried to QC, I got this error:
Any input on this?
 
> annotation <- GetGRangesFromEnsDb(ensdb = EnsDb.Hsapiens.v86)
Error in GetGRangesFromEnsDb(ensdb = EnsDb.Hsapiens.v86) : 
  Please install biovizBase
https://www.bioconductor.org/packages/biovizBase/
> seqlevelsStyle(annotation) <- "UCSC"
Error in (function (classes, fdef, mtable)  : 
  unable to find an inherited method for function 'seqinfo' for signature '"standardGeneric"'
